File: bannerlayout/src/main/java/com/wkz/bannerlayout/utils/TransformerUtils.java
package com.wkz.bannerlayout.utils;

import android.support.annotation.NonNull;

import com.wkz.bannerlayout.animation.AccordionTransformer;
import com.wkz.bannerlayout.animation.BackgroundToForegroundTransformer;
import com.wkz.bannerlayout.animation.BannerTransformer;
import com.wkz.bannerlayout.animation.CubeInTransformer;
import com.wkz.bannerlayout.animation.CubeOutTransformer;
import com.wkz.bannerlayout.animation.DefaultTransformer;
import com.wkz.bannerlayout.animation.DepthPageTransformer;
import com.wkz.bannerlayout.animation.DrawerTransformer;
import com.wkz.bannerlayout.animation.FlipHorizontalTransformer;
import com.wkz.bannerlayout.animation.FlipVerticalTransformer;
import com.wkz.bannerlayout.animation.ForegroundToBackgroundTransformer;
import com.wkz.bannerlayout.animation.RotateDownTransformer;
import com.wkz.bannerlayout.animation.RotateUpTransformer;
import com.wkz.bannerlayout.animation.ScaleInOutTransformer;
import com.wkz.bannerlayout.animation.StackTransformer;
import com.wkz.bannerlayout.animation.TabletTransformer;
import com.wkz.bannerlayout.animation.VerticalTransformer;
import com.wkz.bannerlayout.animation.ZoomInTransformer;
import com.wkz.bannerlayout.animation.ZoomOutPageTransformer;
import com.wkz.bannerlayout.animation.ZoomOutSlideTransformer;
import com.wkz.bannerlayout.animation.ZoomOutTranformer;
import com.wkz.bannerlayout.annotation.PageTransformerMode;

public class TransformerUtils {

    @NonNull
    public static BannerTransformer getTransformer(int type) {
        @PageTransformerMode
        int transformerType = type;
        switch (transformerType) {
            case PageTransformerMode.ANIMATION_ACCORDION:
                return new AccordionTransformer();
            case PageTransformerMode.ANIMATION_BACKGROUND:
                return new BackgroundToForegroundTransformer();
            case PageTransformerMode.ANIMATION_CUBE_IN:
                return new CubeInTransformer();
            case PageTransformerMode.ANIMATION_CUBE_OUT:
                return new CubeOutTransformer();
            case PageTransformerMode.ANIMATION_DEFAULT:
                return new DefaultTransformer();
            case PageTransformerMode.ANIMATION_DEPTH_PAGE:
                return new DepthPageTransformer();
            case PageTransformerMode.ANIMATION_DRAWER:
                return new DrawerTransformer();
            case PageTransformerMode.ANIMATION_FLIPHORIZONTAL:
                return new FlipHorizontalTransformer();
            case PageTransformerMode.ANIMATION_FLIPVERTICAL:
                return new FlipVerticalTransformer();
            case PageTransformerMode.ANIMATION_FOREGROUND:
                return new ForegroundToBackgroundTransformer();
            case PageTransformerMode.ANIMATION_ROTATEDOWN:
                return new RotateDownTransformer();
            case PageTransformerMode.ANIMATION_ROTATEUP:
                return new RotateUpTransformer();
            case PageTransformerMode.ANIMATION_SCALEINOUT:
                return new ScaleInOutTransformer();
            case PageTransformerMode.ANIMATION_STACK:
                return new StackTransformer();
            case PageTransformerMode.ANIMATION_TABLET:
                return new TabletTransformer();
            case PageTransformerMode.ANIMATION_VERTICAL:
                return new VerticalTransformer();
            case PageTransformerMode.ANIMATION_ZOOMIN:
                return new ZoomInTransformer();
            case PageTransformerMode.ANIMATION_ZOOMOUTPAGE:
                return new ZoomOutPageTransformer();
            case PageTransformerMode.ANIMATION_ZOOMOUTSLIDE:
                return new ZoomOutSlideTransformer();
            case PageTransformerMode.ANIMATION_ZOOMOUT:
                return new ZoomOutTranformer();
            default:
                return new DefaultTransformer();
        }
    }
}

File: bannerlayout/src/main/java/com/fphoenixcorneae/bannerlayout/utils/TransformerUtils.kt
package com.fphoenixcorneae.bannerlayout.utils

import androidx.viewpager.widget.ViewPager.PageTransformer
import com.fphoenixcorneae.bannerlayout.annotation.PageTransformerMode
import com.fphoenixcorneae.transformer.*

object TransformerUtils {
    fun getTransformer(@PageTransformerMode type: Int): PageTransformer {
        return when (type) {
            PageTransformerMode.ANIMATION_ACCORDION -> AccordionTransformer()
            PageTransformerMode.ANIMATION_BACKGROUND -> BackgroundToForegroundTransformer()
            PageTransformerMode.ANIMATION_CUBE_IN -> CubeInTransformer()
            PageTransformerMode.ANIMATION_CUBE_OUT -> CubeOutTransformer()
            PageTransformerMode.ANIMATION_DEFAULT -> DefaultTransformer()
            PageTransformerMode.ANIMATION_DEPTH_PAGE -> DepthPageTransformer()
            PageTransformerMode.ANIMATION_DRAWER -> DrawerTransformer()
            PageTransformerMode.ANIMATION_FADE_OUT_FADE_IN -> FadeOutFadeInTransformer()
            PageTransformerMode.ANIMATION_FLIP_HORIZONTAL -> FlipHorizontalTransformer()
            PageTransformerMode.ANIMATION_FLIP_VERTICAL -> FlipVerticalTransformer()
            PageTransformerMode.ANIMATION_FOREGROUND -> ForegroundToBackgroundTransformer()
            PageTransformerMode.ANIMATION_ROTATE_DOWN -> RotateDownTransformer()
            PageTransformerMode.ANIMATION_ROTATE_UP -> RotateUpTransformer()
            PageTransformerMode.ANIMATION_SCALE_IN_OUT -> ScaleInOutTransformer()
            PageTransformerMode.ANIMATION_STACK -> StackTransformer()
            PageTransformerMode.ANIMATION_TABLET -> TabletTransformer()
            PageTransformerMode.ANIMATION_VERTICAL -> VerticalTransformer()
            PageTransformerMode.ANIMATION_ZOOM_IN -> ZoomInTransformer()
            PageTransformerMode.ANIMATION_ZOOM_OUT_PAGE -> ZoomOutPageTransformer()
            PageTransformerMode.ANIMATION_ZOOM_OUT_SLIDE -> ZoomOutSlideTransformer()
            PageTransformerMode.ANIMATION_ZOOM_OUT -> ZoomOutTransformer()
            else -> DefaultTransformer()
        }
    }
}